api/pkg/file/service.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2020 The SODA Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package file
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/emicklei/go-restful"
+	"github.com/micro/go-micro/v2/client"
+	"github.com/opensds/multi-cloud/api/pkg/common"
+	"github.com/opensds/multi-cloud/api/pkg/policy"
+	"github.com/opensds/multi-cloud/backend/proto"
+	"github.com/opensds/multi-cloud/file/proto"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	fileService   = "file"
+	backendService = "backend"
+)
+
+type APIService struct {
+	fileClient   file.FileService
+	backendClient backend.BackendService
+}
+
+func NewAPIService(c client.Client) *APIService {
+	return &APIService{
+		fileClient:   file.NewFileService(fileService, c),
+		backendClient: backend.NewBackendService(backendService, c),
+	}
+}
+
+func UpdateFilter(reqFilter map[string]string, filter map[string]string) error {
+	for k, v := range filter {
+		reqFilter[k] = v
+	}
+	return nil
+}
+
+func WriteError(response *restful.Response, msg string, errCode int, err error) {
+	log.Errorf(msg, err)
+	error := response.WriteError(errCode, err)
+	if error != nil {
+		log.Errorf("Response write status and http error failed: %v\n", err)
+		return
+	}
+}
+
+func (s *APIService) checkBackendExists(ctx context.Context, request *restful.Request, response *restful.Response,
+	backendId string){
+	backendResp, err := s.backendClient.GetBackend(ctx, &backend.GetBackendRequest{Id: backendId})
+	if err != nil {
+		WriteError(response, "Get Backend details failed: %v\n", http.StatusNotFound, err)
+		return
+	}
+	log.Infof("Backend response = [%+v]\n", backendResp)
+}
+
+func (s *APIService) prepareFileShareRequest(request *restful.Request, response *restful.Response) *file.ListFileShareRequest {
+
+	limit, offset, err := common.GetPaginationParam(request)
+	if err != nil {
+		WriteError(response, "Get pagination parameters failed: %v\n", http.StatusInternalServerError, err)
```

Comment:
What kind of error code and error message will return from file share service of AWS, AZure, etc. Suggest to list all those error we used, and be compatible with normal way used by those popular cloud vendor.

Reply:
Yes,
We will have to provide the uniform API Error Handling Codes for Multi-Cloud APIs.
I have opened an issue for handling errors.
https://github.com/sodafoundation/multi-cloud/issues/1009

Reply:
For object service, we have https://github.com/sodafoundation/multi-cloud/blob/master/s3/error/s3errors.go. File service may have different errors.